shm/Cargo.cpp

```diff
@@ -1,24 +1,28 @@
+#include <stdexcept>
+
 #include "Cargo.hpp"
 
-Cargo::Cargo(std::string name, size_t amount, size_t basePrice)
+Cargo::Cargo(const std::string& name, size_t amount, size_t basePrice)
     : name_{name}, amount_{amount}, basePrice_{basePrice} {}
 
-
 Cargo& Cargo::operator+=(size_t amount) {
     amount_ += amount;
     return *this;
```

Comment:
We can assume, that this operator will newer add more cargo than player capacity?